backend/ConstructionManagementSystem/live-settings.txt
"""
Django settings for ConstructionManagementSystem project.

Generated by 'django-admin startproject' using Django 3.2.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""
import os
from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-#+2mjtoz)5vn)ltzvjhheog9a*^qgumqij@$jr-!^u6h*0mapo'
REFRESH_TOKEN_SECRET=SECRET_KEY+'9edc7acb1f3f470496d70374731f3c36'+SECRET_KEY

REST_FRAMEWORK = {
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
    'PAGE_SIZE': 1,
    'DEFAULT_AUTHENTICATION_CLASSES': (
        'project.api.login.authentication.SafeJWTAuthentication',
    ),
    # 'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAuthenticated',),
    # 'DEFAULT_PARSER_CLASSES': (
    #     'rest_framework.parsers.FormParser',
    #     'rest_framework.parsers.MultiPartParser'
    # ),
}

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

DATA_UPLOAD_MAX_MEMORY_SIZE = 262144000

ALLOWED_HOSTS = ['*']
AUTH_USER_MODEL = 'core.User'
LOGOUT_REDIRECT_URL = "/login/"
LOGIN_REDIRECT_URL = "/"

SITE_ID = 1

# Application definition

INSTALLED_APPS = [
    'app_admin',
    'dashboard.apps.AppAdminConfig',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.humanize',
    'django_select2',
    'daterange_filter',
    'corsheaders',
    'rest_framework',
    'rest_framework_simplejwt.token_blacklist',
    'mptt',
    'notifications',
    'colorfield',
    'vendor_company',  
    'core',
    'organization',
    'dashboard.apps.DashboardConfig',
    'project.apps.ProjectConfig',
]

CORE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
TEMPLATE_DIR = os.path.join(CORE_DIR, "templates")  # ROOT dir for templates

MIDDLEWARE = [
    'corsheaders.middleware.CorsMiddleware',
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]


# CORS_ORIGIN_ALLOW_ALL = True # If this is used then `CORS_ORIGIN_WHITELIST` will not have any effect
# CORS_ALLOW_CREDENTIALS = True

# SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
# CSRF_TRUSTED_ORIGINS = ['lbr.askariqa.com','live.askariqa.com','ld.askariqa.com','askariqa.com']

# CORS_REPLACE_HTTPS_REFERER = True

# CSRF_COOKIE_DOMAIN = 'askariqa.com'

SESSION_COOKIE_SECURE = False
CSRF_COOKIE_SECURE = False

# CORS_ALLOW_METHODS = [
#     "DELETE",
#     "GET",
#     "OPTIONS",
#     "PATCH",
#     "POST",
#     "PUT",
# ]

ROOT_URLCONF = 'ConstructionManagementSystem.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [TEMPLATE_DIR],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
            # 'loaders': [
            #     'django.template.loaders.app_directories.Loader',
            # ],
        },
    },
]

WSGI_APPLICATION = 'ConstructionManagementSystem.wsgi.application'
ASGI_APPLICATION = 'ConstructionManagementSystem.asgi.application'


# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

SESSION_COOKIE_SECURE = False
DATABASES = {

    'default': {

        'ENGINE': 'django.db.backends.postgresql_psycopg2',

        'NAME': 'aqa',

        'USER': 'postgres',

        'PASSWORD': 'd0XjMR7Bx6gzqF8',
        #'PASSWORD': 'admin321',

        'HOST': '95.217.7.86',

        'PORT': '5432',

    }


    # 'quill_bcms': {

    #     'ENGINE': 'django.db.backends.postgresql_psycopg2',

    #     'NAME': 'quill_bcms',

    #     'USER': 'postgres',

    #     'PASSWORD': 'qEcUQA7RTTCamV',
    #     #'PASSWORD': 'admin321',

    #     'HOST': 'localhost',

    #     'PORT': '5432',

    # },
}

DATABASE_ROUTERS = ('core.dbrouters.MyDBRouter',)

# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Karachi'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'

# Add these new lines
STATICFILES_DIRS = (os.path.join(BASE_DIR, 'static'),
)

STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')



# STATIC_ROOT = os.path.join(CORE_DIR, 'staticfiles')
# STATIC_URL = '/static/'

# # Extra places for collectstatic to find static files.
# STATICFILES_DIRS = (
#     os.path.join(CORE_DIR, 'static'),
# )

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# LOGGING = {
#     'version': 1,
#     'disable_existing_loggers': False,
#     'formatters': {
#         'standard': {
#             'format' : "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
#             'datefmt' : "%d/%b/%Y %H:%M:%S"
#         },
#     },
#     'handlers': {
#         'default': {
#         'level':'DEBUG',
#         'class':'logging.handlers.RotatingFileHandler',
#         'filename': os.path.join(BASE_DIR, '../request_logs/debug.log'),
#         'maxBytes': 1024*1024*1, # 5 MB
#         'backupCount': 30,
#         'formatter':'standard',
#     	},
#         'file': {
#             'level': 'ERROR',
#             #'class': 'logging.FileHandler',
#             'class':'logging.handlers.RotatingFileHandler',
#             'filename': os.path.join(BASE_DIR, '../request_logs/error.log'),
#             'maxBytes': 1024*1024*1, # 5 MB
# 	        'backupCount': 30,
# 	        'formatter':'standard',

#         },
#         'request_file': {
#             'level': 'DEBUG',
#             #'class': 'logging.FileHandler',
#             'class':'logging.handlers.RotatingFileHandler',
#             'filename': os.path.join(BASE_DIR, '../request_logs/request_file.log'),
#             'maxBytes': 1024*1024*1, # 5 MB
#             'backupCount': 30,
#             #'formatter':'standard',

#         },
#     },
#     'loggers': {
#         'django': {
#             'handlers': ['file'],
#             'level': 'ERROR',
#             'propagate': True,
#         },
#         'django.request': {
#             'handlers': ['request_file'],
#             'level': 'DEBUG',  # change debug level as appropiate
#             'propagate': False,
#         },
#     },
# }

# Notification Broadcaster Settings

# NOTIFICATIONS_WEBSOCKET_URL_PARAM = 'r'ws/notification/(?P<room_name>\w+)/$'

# CHANNEL_LAYERS = {
#     'default': {
#         'BACKEND': 'channels_redis.core.RedisChannelLayer',
#         'CONFIG': {
#             "hosts": [('127.0.0.1', 6379)],
#         },
#     },
# }

CACHES = {
    "default": {
        "BACKEND": "django_redis.cache.RedisCache",
        "LOCATION": "redis://127.0.0.1:6379/16",  # Use the correct database number, in this example it's 1
        "OPTIONS": {
            "CLIENT_CLASS": "django_redis.client.DefaultClient",
        }
    }
}


# The default meta precedence order
IPWARE_META_PRECEDENCE_ORDER = (
 'HTTP_X_FORWARDED_FOR', 'X_FORWARDED_FOR',  # <client>, <proxy1>, <proxy2>
 'HTTP_CLIENT_IP',
 'HTTP_X_REAL_IP',
 'HTTP_X_FORWARDED',
 'HTTP_X_CLUSTER_CLIENT_IP',
 'HTTP_FORWARDED_FOR',
 'HTTP_FORWARDED',
 'HTTP_VIA',
 'REMOTE_ADDR',
)


NOTIFICATIONS_WEBSOCKET_URL_PARAM = 'notifications'

# from notifications.default_settings import *
























































































backend/ConstructionManagementSystem/settings.py
"""
Django settings for ConstructionManagementSystem project.

Generated by 'django-admin startproject' using Django 3.2.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""
import os
from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-#+2mjtoz)5vn)ltzvjhheog9a*^qgumqij@$jr-!^u6h*0mapo'
REFRESH_TOKEN_SECRET=SECRET_KEY+'9edc7acb1f3f470496d70374731f3c36'+SECRET_KEY

REST_FRAMEWORK = {
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
    'PAGE_SIZE': 1,
    'DEFAULT_AUTHENTICATION_CLASSES': (
        'project.api.login.authentication.SafeJWTAuthentication',
    ),
    # 'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAuthenticated',),
    # 'DEFAULT_PARSER_CLASSES': (
    #     'rest_framework.parsers.FormParser',
    #     'rest_framework.parsers.MultiPartParser'
    # ),
}

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

DATA_UPLOAD_MAX_MEMORY_SIZE = 262144000

ALLOWED_HOSTS = ['*']
AUTH_USER_MODEL = 'core.User'
LOGOUT_REDIRECT_URL = "/login/"
LOGIN_REDIRECT_URL = "/"

SITE_ID = 1

# Application definition

INSTALLED_APPS = [
    'app_admin',
    'dashboard.apps.AppAdminConfig',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.humanize',
    'django_select2',
    'daterange_filter',
    'corsheaders',
    'rest_framework',
    'rest_framework_simplejwt.token_blacklist',
    'mptt',
    'notifications',
    'colorfield',
    'vendor_company',  
    'core',
    'organization',
    'dashboard.apps.DashboardConfig',
    'project.apps.ProjectConfig',
]

CORE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
TEMPLATE_DIR = os.path.join(CORE_DIR, "templates")  # ROOT dir for templates

MIDDLEWARE = [
    'corsheaders.middleware.CorsMiddleware',
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]


# CORS_ORIGIN_ALLOW_ALL = True # If this is used then `CORS_ORIGIN_WHITELIST` will not have any effect
# CORS_ALLOW_CREDENTIALS = True

# SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
# CSRF_TRUSTED_ORIGINS = ['lbr.askariqa.com','live.askariqa.com','ld.askariqa.com','askariqa.com']

# CORS_REPLACE_HTTPS_REFERER = True

# CSRF_COOKIE_DOMAIN = 'askariqa.com'

SESSION_COOKIE_SECURE = False
CSRF_COOKIE_SECURE = False

# CORS_ALLOW_METHODS = [
#     "DELETE",
#     "GET",
#     "OPTIONS",
#     "PATCH",
#     "POST",
#     "PUT",
# ]

ROOT_URLCONF = 'ConstructionManagementSystem.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [TEMPLATE_DIR],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
            # 'loaders': [
            #     'django.template.loaders.app_directories.Loader',
            # ],
        },
    },
]

WSGI_APPLICATION = 'ConstructionManagementSystem.wsgi.application'
ASGI_APPLICATION = 'ConstructionManagementSystem.asgi.application'


# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

SESSION_COOKIE_SECURE = False
DATABASES = {

    'default': {

        'ENGINE': 'django.db.backends.postgresql_psycopg2',

       # 'NAME': 'aqa',
       #'NAME': 'aqa_22_11_23',
       # 'NAME': 'aqa_8_1_24',
       #'NAME': 'aqa_10_6_24',# This is the latest local database
        'NAME': 'aqa_28_4_25',
        #'NAME': 'aqa_21_10_24',
        'USER': 'postgres',

        'PASSWORD': 'HfD3rpBztnqTh0k',
       # 'PASSWORD': 'iMbHKeNfntavxeiEfKPk',
        #'PASSWORD': 'rehan@12345',
        
        
        #'PASSWORD': 'admin321',

        #'HOST': '95.217.7.86',
        'HOST': '127.0.0.1',
        #'HOST': '95.216.170.137',
       
        'PORT': '5432',

    }

    # 'quill_bcms': {

    #     'ENGINE': 'django.db.backends.postgresql_psycopg2',

    #     'NAME': 'quill_bcms',

    #     'USER': 'postgres',

    #     'PASSWORD': 'qEcUQA7RTTCamV',
    #     #'PASSWORD': 'admin321',

    #     'HOST': 'localhost',

    #     'PORT': '5432',

    # },
}

DATABASE_ROUTERS = ('core.dbrouters.MyDBRouter',)

# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'Asia/Karachi'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'

# Add these new lines
STATICFILES_DIRS = (os.path.join(BASE_DIR, 'static'),
)

STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')



# STATIC_ROOT = os.path.join(CORE_DIR, 'staticfiles')
# STATIC_URL = '/static/'

# # Extra places for collectstatic to find static files.
# STATICFILES_DIRS = (
#     os.path.join(CORE_DIR, 'static'),
# )

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# LOGGING = {
#     'version': 1,
#     'disable_existing_loggers': False,
#     'formatters': {
#         'standard': {
#             'format' : "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
#             'datefmt' : "%d/%b/%Y %H:%M:%S"
#         },
#     },
#     'handlers': {
#         'default': {
#         'level':'DEBUG',
#         'class':'logging.handlers.RotatingFileHandler',
#         'filename': os.path.join(BASE_DIR, '../request_logs/debug.log'),
#         'maxBytes': 1024*1024*1, # 5 MB
#         'backupCount': 30,
#         'formatter':'standard',
#     	},
#         'file': {
#             'level': 'ERROR',
#             #'class': 'logging.FileHandler',
#             'class':'logging.handlers.RotatingFileHandler',
#             'filename': os.path.join(BASE_DIR, '../request_logs/error.log'),
#             'maxBytes': 1024*1024*1, # 5 MB
# 	        'backupCount': 30,
# 	        'formatter':'standard',

#         },
#         'request_file': {
#             'level': 'DEBUG',
#             #'class': 'logging.FileHandler',
#             'class':'logging.handlers.RotatingFileHandler',
#             'filename': os.path.join(BASE_DIR, '../request_logs/request_file.log'),
#             'maxBytes': 1024*1024*1, # 5 MB
#             'backupCount': 30,
#             #'formatter':'standard',

#         },
#     },
#     'loggers': {
#         'django': {
#             'handlers': ['file'],
#             'level': 'ERROR',
#             'propagate': True,
#         },
#         'django.request': {
#             'handlers': ['request_file'],
#             'level': 'DEBUG',  # change debug level as appropiate
#             'propagate': False,
#         },
#     },
# }

# Notification Broadcaster Settings

# NOTIFICATIONS_WEBSOCKET_URL_PARAM = 'r'ws/notification/(?P<room_name>\w+)/$'

# CHANNEL_LAYERS = {
#     'default': {
#         'BACKEND': 'channels_redis.core.RedisChannelLayer',
#         'CONFIG': {
#             "hosts": [('127.0.0.1', 6379)],
#         },
#     },
# }

CACHES = {
    "default": {
        "BACKEND": "django_redis.cache.RedisCache",
        "LOCATION": "redis://127.0.0.1:6379/16",  # Use the correct database number, in this example it's 1
        "OPTIONS": {
            "CLIENT_CLASS": "django_redis.client.DefaultClient",
        }
    }
}


# The default meta precedence order
IPWARE_META_PRECEDENCE_ORDER = (
 'HTTP_X_FORWARDED_FOR', 'X_FORWARDED_FOR',  # <client>, <proxy1>, <proxy2>
 'HTTP_CLIENT_IP',
 'HTTP_X_REAL_IP',
 'HTTP_X_FORWARDED',
 'HTTP_X_CLUSTER_CLIENT_IP',
 'HTTP_FORWARDED_FOR',
 'HTTP_FORWARDED',
 'HTTP_VIA',
 'REMOTE_ADDR',
)


NOTIFICATIONS_WEBSOCKET_URL_PARAM = 'notifications'

# from notifications.default_settings import *






















































































